src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,32 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
+    private static final int SLEEP_TIME = 200;
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() ->
+                cache.computeIfAbsent(userId, this::getUserData), executor);
+    }
+
+    private UserData getUserData(String userId) {
+        try {
+            TimeUnit.MILLISECONDS.sleep(SLEEP_TIME);
+        } catch (InterruptedException e) {
+            throw new IllegalStateException(e);
```

Comment:
Interrupted exceptions should be handled properly. In this case, you are rethrowing it as an unchecked exception, which may not be the best approach. Consider either handling the interruption more gracefully or resetting the interrupt flag with `Thread.currentThread().interrupt();` before throwing the exception.